src/passes/Memory64Lowering.cpp

```diff
@@ -91,6 +91,11 @@ struct Memory64Lowering : public WalkerPass<PostWalker<Memory64Lowering>> {
   }
 
   void visitMemory(Memory* memory) {
+    for (auto& segment : memory->segments) {
+      auto c = segment.offset->cast<Const>();
```

Comment:
`auto` is copy by default, so while `&` is often needed since it makes a semantic difference, `*` never makes any semantic difference. Since the objective of `auto` is to remove type-clutter from the code, adding the type partially back in seems unnecessary to me. But I'll add it.